src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py

```diff
@@ -206,1004 +209,157 @@ def create_dir(directory):
-        else:
-            gs.fatal(
-                _(
-                    "{} is an invalid hash function. " "Please Enter MD5 or SHA256"
-                ).format(hash_function)
-            )
-
-    return readable_hash
-
-
-def download_gcs_file(url, destination, checksum_function, checksum):
-    """Downloads a single file from GCS and performs checksumming."""
-    # if file exists, check if checksum is ok, download again otherwise
-    if os.path.isfile(destination):
-        sum_existing = get_checksum(destination, checksum_function)
-        if sum_existing == checksum:
-            return 0
-    try:
-        r_file = requests.get(url, allow_redirects=True)
-        with open(destination, "wb") as dest:
-            dest.write(r_file.content)
-        sum_dl = get_checksum(destination, checksum_function)
-        if sum_dl.lower() != checksum.lower():
-            gs.verbose(_("Checksumming not successful for {}").format(destination))
-            return 2
-        return 0
-
-    except Exception as e:
-        gs.verbose(_("There was a problem downloading {}").format(url))
-        return 1
+    normalized_datetime = datetime.fromisoformat(datetime_str)
+    if normalized_datetime.tzinfo is None:
+        return normalized_datetime.strftime("%Y-%m-%dT%H:%M:%S")
+    return normalized_datetime.astimezone(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S")
+
+
+PRODUCTTYPE_MAP = {
+    "S2MSI1C": "S2_MSI_L1C",
+    "S2MSI2A": "S2_MSI_L2A",
+    "OCN": "S1_SAR_OCN",
+    "GRD": "S1_SAR_GRD",
+    "SLC": "S1_SAR_SLC",
+    "S3OL1EFR": "S3_EFR",
+    "S3OL1ERR": "S3_ERR",
+    "S3SL1RBT": "S3_SLSTR_L1RBT",
+    "S3OL2WFR": "S3_OLCI_L2WFR",
+    "S3OL2WRR": "S3_OLCI_L2WRR",
+    "S3OL2LFR": "S3_OLCI_L2LFR",
+    "S3OL2LRR": "S3_OLCI_L2LRR",
+    "S3SL2LST": "S3_SLSTR_L2LST",
+    "S3SL2FRP": "S3_SLSTR_L2FRP",
+    "S3SR2LAN": "S3_LAN",
+    "S3SY2SYN": "S3_SY_SYN",
+    "S3SY2VGP": "S3_SY_VGP",
+    "S3SY2VG1": "S3_SY_VG1",
+    "S3SY2V10": "S3_SY_V10",
+    "S3SY2AOD": "S3_SY_AOD",
+    "S3OL1RAC": "S3_RAC - SARA/WEKEO",
+    "S3OL1SAC": "DEPRECATED",  # Can not be found anywhere within EODAG
+}
+
+REVERSE_PRODUCTTYPE_MAP = {
+    "S2_MSI_L1C": "S2MSI1C",
+    "S2_MSI_L2A": "S2MSI2A",
+    "S1_SAR_OCN": "OCN",
+    "S1_SAR_GRD": "GRD",
+    "S1_SAR_SLC": "SLC",
+    "S3_EFR": "S3OL1EFR",
+    "S3_ERR": "S3OL1ERR",
+    "S3_SLSTR_L1RBT": "S3SL1RBT",
+    "S3_OLCI_L2WFR": "S3OL2WFR",
+    "S3_OLCI_L2WRR": "S3OL2WRR",
+    "S3_OLCI_L2LFR": "S3OL2LFR",
+    "S3_OLCI_L2LRR": "S3OL2LRR",
+    "S3_SLSTR_L2LST": "S3SL2LST",
+    "S3_SLSTR_L2FRP": "S3SL2FRP",
+    "S3_LAN": "S3SR2LAN",
+    "S3_SY_SYN": "S3SY2SYN",
+    "S3_SY_VGP": "S3SY2VGP",
+    "S3_SY_VG1": "S3SY2VG1",
+    "S3_SY_V10": "S3SY2V10",
+    "S3_SY_AOD": "S3SY2AOD",
+    "S3_RAC - SARA/WEKEO": "S3OL1RAC",
+    "S3OL1SAC": "DEPRECATED",  # Can not be found anywhere within EODAG
+}
+
+CLOUDCOVER_PRODUCTS = ["S2MSI1C", "S2MSI2A"]
```

Comment:
S1 products do not have cloudcover, that's right because they are SAR data. But S3 is optic afaik, don't they have cloud cover property?? Just checking, haven't used S3.

Reply:
After some testing, these are the Sentinel-3 products that have cloud coverage percentage:
* S3SY2SYN
* S3SY2VGP
* S3SY2VG1
* S3SY2V10
* S3SY2AOD

The rest do have the cloud coverage attribute but it is always 0.